cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/deploy/InMemoryProgramRunDispatcher.java

```diff
@@ -278,11 +325,6 @@ protected Program createProgram(CConfiguration cConf, ProgramRunner programRunne
         if (isTetheredRun) {
           // This hack is here because the program jar won't actually exist in a tethered run
           // TODO: (CDAP-19150) remove hack once Location is removed form ArtifactDetail
```

Comment:
nit - can you also move this comment the place where this logic happens?

Reply:
fixed